### PR TITLE
8257467: [TESTBUG] -Wdeprecated-declarations is reported at sigset() in exesigtest.c

### PR DESCRIPTION
**Pull request by YaSuenag**
`sigset()` is deprecated, and `__attribute_deprecated_msg__` has been [set to the declaration in glibc](https://sourceware.org/git/?p=glibc.git;a=commit;f=signal/signal.h;h=02802fafcf6e11ea3f998f685035ffe568dfddeb)

We can see the warning on `make test-image` as below:

* Fedora 33 x86_64
    * gcc: gcc-10.2.1-6.fc33.x86_64
    * glibc: glibc-2.32-2.fc33.x86_64

```
/home/ysuenaga/github-forked/jdk/test/hotspot/jtreg/runtime/signal/exesigtest.c: In function 'setSignalHandler':
/home/ysuenaga/github-forked/jdk/test/hotspot/jtreg/runtime/signal/exesigtest.c:245:9: error: 'sigset' is deprecated: Use the signal and sigprocmask functions instead [-Werror=deprecated-declarations]
  245 | sigset(signal_num, handler);
      | ^~~~~~
In file included from /home/ysuenaga/github-forked/jdk/test/hotspot/jtreg/runtime/signal/exesigtest.c:25:
/usr/include/signal.h:353:23: note: declared here
  353 | extern __sighandler_t sigset (int __sig, __sighandler_t __disp) __THROW
      | ^~~~~~
cc1: all warnings being treated as errors
gmake[3]: *** [test/JtregNativeHotspot.gmk:1525: /home/ysuenaga/github-forked/jdk/build/linux-x86_64-server-fastdebug/support/test/hotspot/jtreg/native/support/exesigtest/exesigtest.o] Error 1
gmake[3]: *** Waiting for unfinished jobs....
gmake[2]: *** [make/Main.gmk:612: build-test-hotspot-jtreg-native] Error 2
```

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257467](https://bugs.openjdk.java.net/browse/JDK-8257467): [TESTBUG] -Wdeprecated-declarations is reported at sigset() in exesigtest.c

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 19c35bba9c78c54fa63ff6085fe1d8281d0d553a
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1529/head:pull/1529`
`$ git checkout pull/1529`

### CODE REVIEW

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
I would suggest we just delete the sigset usage as it is obsolete. The whole point of this test is to test signal chaining for the VM and sigset is undefined for a multi-threaded process (POSIX spec.) The use of sigaction suffices IMO.
Thanks,
David

**Comment by YaSuenag** [comment]
It's better if we can remove `sigset()` from signal handler tests. So I did that in new commit. Could you review again?
This commit passed hotspot/jtreg/runtime/signal tests.

File: test/hotspot/jtreg/runtime/signal/README [hunk 1]
```diff
@@ -37,7 +37,7 @@ There is one more scenario called 'nojvm'.
 In this case no jvm is created, so pure signal testing is done.
 
 Signal handlers don't do anything, so the only fact that signal handler was called is checked.
-Also 2 different ways of setting signal handlers are tested: sigaction, sigset.
+Also 2 different ways of setting signal handlers are tested: sigaction.
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/signal/README:40]: There is only one way now so delete this sentence.

File: test/hotspot/jtreg/runtime/signal/SigTestDriver.java [hunk 2]
```diff
@@ -93,7 +93,7 @@ public static void main(String[] args) {
 
         boolean passed = true;
 
-        for (String mode : new String[] {"sigset", "sigaction"}) {
+        for (String mode : new String[] {"sigaction"}) {
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/signal/SigTestDriver.java:96]: You don't need the loop any more. You don't need to pass a "mode" any more.

File: test/hotspot/jtreg/runtime/signal/exesigtest.c [hunk 3]
```diff
@@ -144,7 +144,7 @@ boolean isSupportedSigScenario ()
 
 boolean isSupportedSigMode ()
 {
-    if ( (!strcmp(mode, "sigset")) || (!strcmp(mode, "sigaction")) )
+    if ( !strcmp(mode, "sigaction") )
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/signal/exesigtest.c:147]: You can delete the whole "mode" logic.

**Review by dholmes-ora — COMMENTED** [review]
Just eradicate "mode" completely.

Thanks,
David

**Comment by YaSuenag** [comment]
@dholmes-ora Thanks for your comment! I fixed them in new commit. Could you check again?
This commit passed hotspot/jtreg/runtime/signal tests.

**Comment by YaSuenag** [comment]
Linux x64 tier1 test (runtime/handshake/HandshakeWalkStackTest.java) failed, but it does not seem to cause by this change.

**Review by dholmes-ora — APPROVED** [review]
Looks good to me!

Perhaps @tstuefe could also take a look?

Thanks,
David

**Comment by tstuefe** [comment]
Hi,

sorry for chiming in too late, I did not see this before. 

I would not remove the sigset testing from this test unless we also remove the sigset chaining capability from libjsig. Otherwise we just ship untested functionality.

So I'd say either go the full way and remove sigset() chaining from src/java.base/unix/native/libjsig/jsig.c. But that probably needs a CSR as well. Or we just go with Yasumasa's initial patch, just switching off the warning.

Cheers, Thomas

**Comment by dholmes-ora** [comment]
Hi Thomas (@tstuefe ),
Thanks for chiming in. I was ready to agree with you but something was nagging at me. Sure enough looking closer signal chaining supports three modes: sigset, sigaction and signal. But the test only tests sigset and sigaction - so we already have untested functionality. And that was because signal() use was broken and we removed it under:

https://bugs.openjdk.java.net/browse/JDK-8165192

back in 2016.

So we do have a precedence (perhaps a bad one) for removing test code without removing functional code.

But I'm okay with going back to Yasumasa's original proposal to just fix the warning and filing a RFE to deprecate, then remove sigset and signal from the signal chaining mechanism.

Thanks,
David
-----

**Comment by YaSuenag** [comment]
> But I'm okay with going back to Yasumasa's original proposal to just fix the warning

Reverted commits for removing sigset tests.

> and filing a RFE to deprecate, then remove sigset and signal from the signal chaining mechanism.

I can file it, but I wonder how we mark deprecate to sigset and signal? `#pragma` in each compilers?
Or can we remove them immediately after CSR is approved?

**Comment by tstuefe** [comment]
> > But I'm okay with going back to Yasumasa's original proposal to just fix the warning
> 
> Reverted commits for removing sigset tests.
> 
> > and filing a RFE to deprecate, then remove sigset and signal from the signal chaining mechanism.
> 
> I can file it, but I wonder how we mark deprecate to sigset and signal? `#pragma` in each compilers?
> Or can we remove them immediately after CSR is approved?

I think the official Oracle doc for signal chaining would have to be adapted (https://docs.oracle.com/javase/9/vm/signal-chaining.htm).

**Review by tstuefe — APPROVED** [review]
LGTM

**Comment by YaSuenag** [comment]
> > > and filing a RFE to deprecate, then remove sigset and signal from the signal chaining mechanism.
> > 
> > 
> > I can file it, but I wonder how we mark deprecate to sigset and signal? `#pragma` in each compilers?
> > Or can we remove them immediately after CSR is approved?
> 
> I think the official Oracle doc for signal chaining would have to be adapted (https://docs.oracle.com/javase/9/vm/signal-chaining.htm).

Thanks @tstuefe !
Anyway, I will integrate this change.

**Pull request closed without merging**